homeassistant/components/zha/core/discovery.py

```diff
@@ -66,21 +78,86 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+QUIRKS_ENTITY_META_TO_ENTITY_CLASS = {
+    (
+        Platform.BUTTON,
+        WriteAttributeButtonMetadata,
+        EntityType.CONFIG,
+    ): button.ZHAAttributeButton,
+    (Platform.BUTTON, ZCLCommandButtonMetadata, EntityType.CONFIG): button.ZHAButton,
+    (
+        Platform.BUTTON,
+        ZCLCommandButtonMetadata,
+        EntityType.DIAGNOSTIC,
+    ): button.ZHAButton,
+    (
+        Platform.BINARY_SENSOR,
+        BinarySensorMetadata,
+        EntityType.CONFIG,
+    ): binary_sensor.BinarySensor,
+    (
+        Platform.BINARY_SENSOR,
+        BinarySensorMetadata,
+        EntityType.DIAGNOSTIC,
+    ): binary_sensor.BinarySensor,
+    (
+        Platform.BINARY_SENSOR,
+        BinarySensorMetadata,
+        EntityType.STANDARD,
+    ): binary_sensor.BinarySensor,
+    (Platform.SENSOR, ZCLEnumMetadata, EntityType.DIAGNOSTIC): sensor.EnumSensor,
+    (Platform.SENSOR, ZCLEnumMetadata, EntityType.STANDARD): sensor.EnumSensor,
+    (Platform.SENSOR, ZCLSensorMetadata, EntityType.DIAGNOSTIC): sensor.Sensor,
+    (Platform.SENSOR, ZCLSensorMetadata, EntityType.STANDARD): sensor.Sensor,
+    (Platform.SELECT, ZCLEnumMetadata, EntityType.CONFIG): select.ZCLEnumSelectEntity,
+    (
+        Platform.SELECT,
+        ZCLEnumMetadata,
+        EntityType.DIAGNOSTIC,
+    ): select.ZCLEnumSelectEntity,
+    (
+        Platform.NUMBER,
+        NumberMetadata,
+        EntityType.CONFIG,
+    ): number.ZHANumberConfigurationEntity,
+    (Platform.NUMBER, NumberMetadata, EntityType.DIAGNOSTIC): number.ZhaNumber,
+    (Platform.NUMBER, NumberMetadata, EntityType.STANDARD): number.ZhaNumber,
+    (
+        Platform.SWITCH,
+        SwitchMetadata,
+        EntityType.CONFIG,
+    ): switch.ZHASwitchConfigurationEntity,
+    (Platform.SWITCH, SwitchMetadata, EntityType.STANDARD): switch.Switch,
+}
```

Comment:
Some of the entity classes set the category for HA (e.g. `ZHASwitchConfigurationEntity`, others like `ZHAAttributeButton` do not).
I'm not seeing anywhere where `_attr_entity_category` is set to the `entity_type` of zigpy `EntityMetadata`. Should that be done in the base ZHA entity class?

Reply:
not in the base intentionally. the type is only used in the matching logic here atm and most classes set the category appropriately. This needs to be addressed subsequently when it is safe to set it in the base.